app/controllers/puppetclasses_controller.rb

```diff
@@ -85,4 +86,8 @@ def action_permission
         super
     end
   end
+
+  def fetch_associations
+    ActiveRecord::Associations::Preloader.new(@puppetclass.lookup_keys, [:lookup_values]).run
```

Comment:
sometimes I wonder why I even bother disagreeing with you. you're right, as you usually are :smile: